lib/travis/api/app/endpoint.rb

```diff
@@ -19,6 +19,7 @@ class Endpoint < Base
 
     before do
       halt 406 if accept_version == 'v2.1' && ENV['DISABLE_V2_1']
+      halt 403 if force_auth? && !authenticated?
```

Comment:
Just ran into a funny issue in an incognito window on our test instance: when you're on `/auth` and you click "Sign in with GitHub" it directs you to `/api/auth/handshake?redirect_uri=[github server]`

but because this checks to see if you're authenticated (but you're not, which is why you're authenticating), you get this: 
![image](https://user-images.githubusercontent.com/47749/43004898-6b7212fe-8c31-11e8-9c0f-9872a47c589b.png)

Could be something else going on here, but I'm not seeing where the `/api/auth` endpoint skips the 403 auth check.

Reply:
Ah how annoying. Yeah this makes sense though. I wonder how I didn't spot it. Okay fixing now.

Reply:
Yeah! I was curious how I didn't catch it last night when I did all my testing, I guess I didn't do it with a logged out window now that I think of it? 🤷‍♂️

Reply:
Ok, latest commit should fix that. Let me know how you get on.

I don't quite follow @acnagy's comment in the other issue. Was something else not behaving correctly?